### PR TITLE
Create interface and basic framework for Database Manager

### PR DESCRIPTION
**Pull request by cel243**
I've created the interface for DataReadManager and DataUpdateManager, as well as creating the infrastructure for easily faking out these objects. No functionality has been implemented yet.

### CODE REVIEW

File: src/main/java/com/google/impactdashboard/configuration/Configuration.java [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+   * When set to true, a call to DataReadManagerFactory.create() will return a 
+   * fake of DataReadManager for testing purposes. 
+   */
+  public static boolean useFakeDataReadManager = false; 
+
+  /** 
+   * When set to true, a call to DataUpdateManagerFactory.create() will return a 
+   * fake of DataUpdateManager for testing purposes. 
+   */
+  public static boolean useFakeDataUpdateManager = false;
```
Comment (peterpie123) [inline src/main/java/com/google/impactdashboard/configuration/Configuration.java:15]: Why is `userFakeDataUpdateManager` here twice?

Comment (peterpie123) [reply src/main/java/com/google/impactdashboard/configuration/Configuration.java:15]: Oops nevermind -- ignore me :)

File: src/main/java/com/google/impactdashboard/configuration/Configuration.java [hunk 2]
```diff
@@ -0,0 +1,16 @@
+package com.google.impactdashboard.configuration;
+
+public class Configuration {
```
Comment (davidgaskins) [inline src/main/java/com/google/impactdashboard/configuration/Configuration.java:3]: Javadoc please :)

File: src/main/java/com/google/impactdashboard/database_manager/data_read/DataReadManager.java [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+import java.util.List;
+import java.util.Map;
+
+/** Interface for retrieving data from the database. */
+public interface DataReadManager {
+  /** 
+   *  Returns a list containing identifying information for all projects in the 
+   *  IAM Bindings table.  
+   */
+  public List<ProjectIdentification> listProjects();
```
Comment (davidgaskins) [inline src/main/java/com/google/impactdashboard/database_manager/data_read/DataReadManager.java:14]: This code appears to be synchronous, but the design doc mentions that this call can be made asynchronously, is the asynchronous aspect happening in a future edition?

File: src/main/java/com/google/impactdashboard/configuration/Configuration.java (hunk 2 above)
Comment (cel243) [reply src/main/java/com/google/impactdashboard/configuration/Configuration.java:4]: Added!

File: src/main/java/com/google/impactdashboard/database_manager/data_read/DataReadManager.java (hunk 3 above)
Comment (cel243) [reply src/main/java/com/google/impactdashboard/database_manager/data_read/DataReadManager.java:14]: I'll sync with Ionis and implement this in future versions

**Pull request merged**